server/src/test/java/org/opensearch/rest/action/tiering/HotToWarmTieringResponseTests.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.tiering;
+
+import org.opensearch.action.tiering.HotToWarmTieringResponse;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.test.OpenSearchTestCase;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+public class HotToWarmTieringResponseTests extends OpenSearchTestCase {
```

Comment:
* There are `AbstractResponseTestCase` and `AbstractRequestTestCase` in `rest-high-level`  package to test out request/response parsing on client side.

* On server side there are `AbstractSerializingTestCase` and `AbstractXContentTestCase` for validating ser/de of response/request